lnwallet/chainfee/filtermanager.go

```diff
@@ -0,0 +1,300 @@
+package chainfee
+
+import (
+	"encoding/json"
+	"fmt"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/rpcclient"
+)
+
+const (
+	// fetchFilterInterval is the interval between successive fetches of
+	// our peers' feefilters.
+	fetchFilterInterval = time.Minute * 5
+
+	// medianWindow is the number of historical values that the
+	// filterManager will track.
+	//
+	// Assuming that each polling interval polls 8 outbound peers, the
+	// value of 48 gives us 6 polling intervals which is ~3 blocks worth of
+	// data.
+	medianWindow = 48
+
+	// minNumFilters is the minimum number of feefilters we need during a
+	// polling interval. If we have fewer than this, we won't consider the
```

Comment:
I think we need to add this check in `healthcheck`, and log warnings here when it happens.

Reply:
NACK - see above.